src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1386,6 +1387,31 @@ internal override IEnumerable<FieldSymbol> GetFieldsToEmit()
                             yield return associatedField;
                         }
                         break;
+                    case SymbolKind.Property:
+                        if (m is SourcePropertySymbolBase propertySymbol)
+                        {
+                            // Ensure the binding is done so we guarantee that we have the BackingField set if necessary.
+                            if (propertySymbol.BackingField is null &&
+                                propertySymbol.CSharpSyntaxNode is BasePropertyDeclarationSyntax propertySyntax)
```

Comment:
>propertySymbol.CSharpSyntaxNode is BasePropertyDeclarationSyntax propertySyntax [](http://example.com/codeflow?start=32&length=79)

Why is this type check important? In general, we are trying to avoid requesting syntax nodes from symbols, unless that is absolutely necessary. Requests like that are likely to bring syntax trees alive and cause reparsing.

Reply:
Fixed.